builder/dockerfile/builder.go

```diff
@@ -104,7 +106,15 @@ func (bm *BuildManager) Build(ctx context.Context, config backend.BuildConfig) (
 	}
 
 	os := ""
-	apiPlatform := system.ParsePlatform(config.Options.Platform)
+	var apiPlatform *specs.Platform
```

Comment:
If we want to get rid of the `else`, this could be

```go
apiPlatform := &specs.Platform

if m, err := platforms.Parse(config.Options.Platform); err == nil {
    sp := m.Spec()
    apiPlatform = &sp
}
```

Also wasn't sure if we should be handling the error here (but could be a breaking change)

Reply:
Hello @thaJeztah , IMO we should return the error to the caller just like other code does in the same function, we need to make sure we get an known `os` and `arch` pair before we take any real action as followed.

Reply:
Sounds good to me 👍

Reply:
Will push an update adding the error handling code, thanks for spotting this 🤝